contracts/strategy/RWAStrategy.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright Fathom 2024
+pragma solidity 0.8.19;
+
+import "./BaseStrategy.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+
+// solhint-disable
+contract RWAStrategy is BaseStrategy {
+    using SafeERC20 for ERC20;
+    using Math for uint256;
+
+    // The minimum amount to send to the manager
+    uint256 public minAmount;
+    uint256 public depositLimit;
+
+    uint256 public totalInvested;
+    uint256 public totalGains;
+    uint256 public totalLosses;
+
+    error ZeroAddress();
+    error AmountTooHigh();
+
+    // The address that we send funds to
+    // and that will buy RWAs with the funds
+    address public immutable managerAddress;
+
+    constructor(
+        address _asset,
+        string memory _name,
+        address _tokenizedStrategyAddress,
+        address _managerAddress,
+        uint256 _minAmount,
+        uint256 _depositLimit
+    ) BaseStrategy(_asset, _name, _tokenizedStrategyAddress) {
+        if (_managerAddress == address(0)) {
+            revert ZeroAddress();
+        }
+        if (_minAmount > ERC20(_asset).totalSupply()) {
+            revert AmountTooHigh();
+        }
+        managerAddress = _managerAddress;
+        minAmount = _minAmount;
+        depositLimit = _depositLimit;
+    }
+
+    function _harvestAndReport() internal override returns (uint256 _totalAssets) {
+        if (!TokenizedStrategy.isShutdown()) {
+            // deposit any loose funds
+            uint256 looseAsset = asset.balanceOf(address(this));
+            if (looseAsset > minAmount) {
+                uint256 _amount = Math.min(looseAsset, availableDepositLimit(address(this)));
+                asset.transfer(managerAddress, _amount);
+                totalInvested += _amount;
+            }
+        }
+        _totalAssets = totalInvested + asset.balanceOf(address(this));
+    }
+
+    function availableDepositLimit(address /*_owner*/) public view override returns (uint256) {
+        // Return the remaining room.
+        return depositLimit - asset.balanceOf(address(this)) > 0 ? depositLimit - asset.balanceOf(address(this)) : 0;
```

Comment:
We will have an underflow error here in case ` asset.balanceOf(address(this)) > depositLimit`
And why are we ignoring `totalInvested` here?

Reply:
And calling `asset.balanceOf(address(this))` twice is not efficient aproach

Reply:
We are ignoring totalInvested here because it doesn't really matters how much is invested for those who are making a new deposit, it's only dependant on the depositLimit and current balance.